Content.Client/_Moffstation/Paper/UI/ForgeSignatureBoundUserInterface.cs

```diff
@@ -0,0 +1,57 @@
+using Content.Shared._Moffstation.Paper.Components;
+using Content.Shared.CCVar;
+using Robust.Client.UserInterface;
+using Robust.Shared.Configuration;
+
+namespace Content.Client._Moffstation.Paper.UI;
+
+/// <summary>
+/// Initializes a <see cref="ForgeSignatureWindow"/> and updates it when new server messages are received.
+/// </summary>
+public sealed class ForgeSignatureBoundUserInterface : BoundUserInterface
+{
+    [Dependency] private readonly IEntityManager _entManager = default!;
+    [Dependency] private readonly IConfigurationManager _cfgManager = default!;
+
+    [ViewVariables]
+    private ForgeSignatureWindow? _window;
+
+    public ForgeSignatureBoundUserInterface(EntityUid owner, Enum uiKey) : base(owner, uiKey)
+    {
+    }
+
+    protected override void Open()
+    {
+        base.Open();
+
+        _window = this.CreateWindow<ForgeSignatureWindow>();
+
+        if (!_entManager.HasComponent<ForgeSignatureComponent>(Owner))
+            return;
+
+        _window.SetMaxSignatureLength(_cfgManager.GetCVar(CCVars.MaxNameLength));
+
+        _window.OnSignatureChanged += OnSignatureChanged;
+        Reload();
+    }
+
+    private void OnSignatureChanged(string newSignature)
+    {
+        if (_entManager.TryGetComponent(Owner, out ForgeSignatureComponent? pen) &&
+            pen.Signature.Equals(newSignature))
+            return;
+
+        if (pen is not { } penComp)
+            return;
+        penComp.Signature = newSignature;
+        SendPredictedMessage(new ForgedSignatureChangedMessage(newSignature));
+    }
+
+    public void Reload()
```

Comment:
```suggestion
    private void Reload()
```

Reply:
IMO just inline this function. The two precondition guards are implied by the one calling location, so this implementation simplifies down to one function call itself.